redash/authentication/google_oauth.py

```diff
@@ -88,7 +88,17 @@ def login():
     next_path = request.args.get('next', url_for("redash.index", org_slug=session.get('org_slug')))
     logger.debug("Callback url: %s", callback)
     logger.debug("Next is: %s", next_path)
-    return google_remote_app().authorize(callback=callback, state=next_path)
+    extra = {}
+    if 'org_slug' in session:
+        org = models.Organization.get_by_slug(session.pop('org_slug'))
+    else:
+        org = current_org
+    if org.google_apps_domains:
+        extra['hd'] = org.google_apps_domains[0]
```

Comment:
This breaks support for multiple Google Apps domains. See #1924 for relevant discussion.

Reply:
Hm, I'm not sure how we could solve this other than hardcoding the hosted domain to chose here using a new setting? What do you think @arikfr?